### ISSUE TITLE
`less -R` regressed handling of "Erase in Line" ANSI code

### ISSUE THREAD

**Issue by snarkmaster** [gwsw/less#166]
To reproduce:

```
echo $'\x1b'[35m$'\x1b'[Khello | less -FR
```

Expected: a purple "hello"
Observed: no output

I tried a few OS & terminal permutations:
 - CentOS8, mosh + tmux, less-530: **good**
 - Fedora33, gnome-terminal, less-551: **good**
 - [Fedora Rawhide](https://src.fedoraproject.org/rpms/less/tree/cfe05cef254627485202014d878ba7f5a4f394c2), both terminals above, less-575: **bad**

I don't see any patches in Rawhide that obviously affect `-R` behavior, so this seems more likely to have regressed on trunk.

If you remove the [Escape in Line `$'\x1b'[K`](https://en.wikipedia.org/wiki/ANSI_escape_code#CSI_(Control_Sequence_Introducer)_sequences) from the input, it works everywhere.

It looks like the bad version of `less` is mistakenly removing the `hello`, plus some other mangling I don't understand:

```
$ echo $'\x1b'[35m$'\x1b'[Khello > XXX
$ script -qefc 'less -FR XXX' /dev/null | xxd
00000000: 0d1b 5b33 356d 1b5b 0d1b 5b4b            ..[35m.[..[K
```

**Comment by snarkmaster** [comment]
This looks to be fixed in Fedora Rawhide 581, but I'll leave this open in case the maintainer wants to flag something in the release notes (I'm not sure whether this was an "in-between releases" bug, or a bug in 563).

Working tree: https://src.fedoraproject.org/rpms/less/tree/6e8fa96224f64d3edaa9091933b6b497b7555c70

**Comment by davide125** [comment]
FYI this doesn't repro with less 581 from https://src.fedoraproject.org/rpms/less/tree/6e8fa96224f64d3edaa9091933b6b497b7555c70, so it was probably fixed somewhere between then and 575.

**Comment by gwsw** [comment]
I believe this is the same as #140.

**Comment by gwsw** [comment]
Fixed in 3ed1315d49d45e8c26db0ae74fa68a9950523d3d
